Sync new docs with pending trunk changes (#57850)
woocommerce/woocommerce: docs/block-development/cart-and-checkout-blocks/checkout-payment-methods/checkout-flow-and-events.md

#### Special States

> The following are booleans exposed via the checkout provider that are independent from each other and checkout statuses but can be used in combination to react to various states in the checkout.

Reviewer: ```suggestion
The following booleans are exposed via the checkout provider. They are independent from each other and checkout statuses, but can be used in combination to react to various states in the checkout.
```

##### **isCalculating**

> `isCalculating` is true when the total is being re-calculated for the order or when a plugin is intentionally disabling the checkout using the `disableCheckoutFor` action (covered in the next section).

Reviewer: ```suggestion
`isCalculating` is true when the total is being re-calculated for the order, or when a plugin is intentionally disabling the checkout using the `disableCheckoutFor` action (covered in the next section).
```

> There are numerous things that might trigger a recalculation of the total: coupons being added or removed, shipping rates updated, shipping rate selected, etc. Instead of having to check each of those individual states, you can reliably just check if this boolean is true (calculating) or false (not calculating).

Reviewer: ```suggestion
There are numerous things that might trigger a recalculation of the total—coupons being added or removed, shipping rates updating, shipping rates being selected, etc. Instead of having to check each of those individual states, you can reliably check if this boolean is true (calculating) or false (not calculating).
```
